Typescript support via new TsProcessor

vetraz opened the pull request:
Typescript support was introduced with new TsProcessor built on the top of the existing JsProcessor. Default pattern was changed to support only `.ts `and `.tsx` files under this processor. TS processing is made via @babel/parser plugin "typescript" which is internal one and does not require additional packages.

No additional configuration required from consumers part.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/instructure/i18nliner-js?pullRequest=1) <br/>All committers have signed the CLA.

Commit:
Typescript support via new TsProcessor

Typescript support was introduced with new TsProcessor built on the top of existing JsProcessor. Default pattern was changed to support only ts and tsx files under this processor. TS processing is made via @babel/parser plugin "typescript" which is internal one and does not require additional packages.

amireh committed:
Release: 1.0.0

republish under @instructure/ scope and do some package housekeeping
